src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -369,6 +370,9 @@ class Microsoft::Terminal::Core::Terminal final :
 
     void _NotifyTerminalCursorPositionChanged() noexcept;
 
+    std::map<std::pair<COLORREF, COLORREF>, COLORREF> _adjustedColorMap;

```

Comment:
This seems like it's going to be expensive -- a std::map from `COLORREF+COLORREF` to `COLORREF` that will *always* contain 256 static things will always require more allocations, hashing, and processing time than a fixed size array (or vector, i guess) that you index directly into.

After all...

```
std::array<COLORREF, 256> _adjustedForegroundColors;
...
// We only do the lookup for colors 0-15; this means we can pack the foreground and background into a single byte with the high bits being the background and the low bits being the foreground.
// Then we can just look it up directly with an O(1) operation
const auto index{ background.GetIndex() << 4 | foreground.GetIndex() };
return _adjustedForegroundColors[index];
```

Reply:
(There is no value in mapping old color pairs to new colors -- the **only** thing you care about is the 16 base colors. They have names (0, 1, 2, 3, 4, 5, ...) which are much shorter than "a pair of two colors (8 bytes)")

Reply:
You can also do a classical multi-dimensional array:
```cpp
std::array<std::array<COLORREF, 16>, 16> variant1;
// or:
COLORREF variant2[16][16];
```